packages/backend/src/open_payments/payment/incoming/service.ts

```diff
@@ -135,7 +141,35 @@ async function createIncomingPayment(
     data: incomingPayment.toData(0n)
   })
 
-  return await addReceivedAmount(deps, incomingPayment, BigInt(0))
+  incomingPayment = await addReceivedAmount(deps, incomingPayment, BigInt(0))
+  if (!deps.config.pollIncomingPaymentCreatedWebhook) {
+    return incomingPayment
+  }
+
+  try {
+    const response = await poll({
+      request: async () =>
+        getApprovedOrCanceledIncomingPayment(deps, { id: incomingPayment.id }),
+      pollingFrequencyMs: deps.config.incomingPaymentCreatedPollFrequency,
+      timeoutMs: deps.config.incomingPaymentCreatedPollTimeout
+    })
+
+    if (response) return response
+    return IncomingPaymentError.ActionNotPerformed
```

Comment:
maybe we have one error for not being approved an another for this timeout?

Reply:
This can be done yes. I did this because end result is basically the same. Action was not performed in this timeframe. I can do `IncomingPaymentError.ActionTimeout` ?

Reply:
We can also keep it like this as long as we have logs for both so we know what happened